hosting/src/components/EditablePlayerRow.jsx

```diff
@@ -0,0 +1,144 @@
+import React, { useState, useEffect } from "react";
+
+const EditablePlayerRow = ({ rank, teams, row }) => {
+  const [name, setName] = useState(row.name);
+  const [goals, setGoals] = useState(row.goalsPredicted);
+  const [goalTeam1, setGoalTeam1] = useState(row.teams.goals[0]);
+  const [goalTeam2, setGoalTeam2] = useState(row.teams.goals[1]);
+  const [outcomeTeam1, setOutcomeTeam1] = useState(row.teams.outcomes[0]);
+  const [outcomeTeam2, setOutcomeTeam2] = useState(row.teams.outcomes[1]);
+  const [outcomeTeam3, setOutcomeTeam3] = useState(row.teams.outcomes[2]);
+  const [isValid, setIsValid] = useState(true);
+  const [isUpdated, setIsUpdated] = useState(false);
+  const id = row.id;
+
+  useEffect(() => {
+    checkIsValid();
+    checkIsUpdated();
+  }, [
+    name,
+    goals,
+    goalTeam1,
+    goalTeam2,
+    outcomeTeam1,
+    outcomeTeam2,
+    outcomeTeam3,
+  ]);
+
+  const onUpdate = async () => {
+    try {
+      await fetch("/api/players", {
+        method: "PUT",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          id,
+          name,
+          goalsPredicted: Number(goals),
+          teams: {
+            goals: [goalTeam1, goalTeam2],
+            outcomes: [outcomeTeam1, outcomeTeam2, outcomeTeam3],
+          },
+        }),
+      });
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  const onDelete = async () => {
+    try {
+      await fetch("/api/players", {
```

Comment:
You could pass on onDelete (and onUpdate) callback from the parent component so it can delete or update the player in there without needing to reload the page once the API call is successful